newrelic/resource_newrelic_synthetics_monitor.go

```diff
@@ -95,158 +94,411 @@ func resourceNewRelicSyntheticsMonitor() *schema.Resource {
+							Required:    true,
+							Description: "",
+						},
+					},
+				},
+			},
+			"enable_screenshot_on_failure_and_script": {
+				Type:        schema.TypeBool,
+				Description: "",
+				Optional:    true,
+			},
+			"custom_headers": {
+				Type:        schema.TypeSet,
+				Description: "",
+				Optional:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:        schema.TypeString,
+							Description: "",
+							Optional:    true,
+						},
+						"value": {
+							Type:        schema.TypeString,
+							Description: "",
+							Optional:    true,
+						},
+					},
+				},
+			},
 		},
 	}
 }
 
-func buildSyntheticsMonitorStruct(d *schema.ResourceData) synthetics.Monitor {
-	monitor := synthetics.Monitor{
-		Name:         d.Get("name").(string),
-		Type:         synthetics.MonitorType(d.Get("type").(string)),
-		Frequency:    uint(d.Get("frequency").(int)),
-		Status:       synthetics.MonitorStatusType(d.Get("status").(string)),
-		SLAThreshold: d.Get("sla_threshold").(float64),
+//func to build the input to create simple browser monitor
+func buildSyntheticsSimpleBrowserMonitor(d *schema.ResourceData) synthetics.SyntheticsCreateSimpleBrowserMonitorInput {
+	inputBase := expandSyntheticsMonitorBase(d)
+
+	simpleBrowserMonitorInput := synthetics.SyntheticsCreateSimpleBrowserMonitorInput{}
+
+	simpleBrowserMonitorInput.AdvancedOptions.CustomHeaders = inputBase.CustomHeaders
+	simpleBrowserMonitorInput.Name = inputBase.Name
+	simpleBrowserMonitorInput.Period = inputBase.Period
+	simpleBrowserMonitorInput.Status = inputBase.Status
+	simpleBrowserMonitorInput.Tags = inputBase.Tags
+	simpleBrowserMonitorInput.Uri = inputBase.URI
+
+	if v, ok := d.GetOk("enable_screenshot_on_failure_and_script"); ok {
+		simpleBrowserMonitorInput.AdvancedOptions.EnableScreenshotOnFailureAndScript = v.(bool)
 	}
 
-	if uri, ok := d.GetOk("uri"); ok {
-		monitor.URI = uri.(string)
+	if v, ok := d.GetOk("locations_public"); ok {
+		simpleBrowserMonitorInput.Locations.Public = expandSyntheticsSimplePublicLocations(v.(*schema.Set).List())
 	}
 
-	locationsRaw := d.Get("locations").(*schema.Set)
-	locations := make([]string, locationsRaw.Len())
-	for i, v := range locationsRaw.List() {
-		locations[i] = fmt.Sprint(v)
+	if v, ok := d.GetOk("locations_private"); ok {
+		simpleBrowserMonitorInput.Locations.Private = expandSyntheticsSimplePrivateLocations(v.(*schema.Set).List())
 	}
 
-	if validationString, ok := d.GetOk("validation_string"); ok {
-		monitor.Options.ValidationString = validationString.(string)
+	if v, ok := d.GetOk("validation_string"); ok {
+		simpleBrowserMonitorInput.AdvancedOptions.ResponseValidationText = v.(string)
 	}
 
-	if verifySSL, ok := d.GetOkExists("verify_ssl"); ok {
-		monitor.Options.VerifySSL = verifySSL.(bool)
+	if v, ok := d.GetOk("verify_ssl"); ok {
+		simpleBrowserMonitorInput.AdvancedOptions.UseTlsValidation = v.(bool)
+	}
```

Comment:
small nitpick: If you put newlines between most of the if's you should do same here.